src/engine/specification.md

```diff
@@ -205,17 +207,19 @@ This structure contains the attributes required to initiate a payload build proc
 
 1. The values `(forkchoiceState.headBlockHash, forkchoiceState.finalizedBlockHash)` of this method call map on the `POS_FORKCHOICE_UPDATED` event of [EIP-3675](https://eips.ethereum.org/EIPS/eip-3675#specification) and **MUST** be processed according to the specification defined in the EIP.
 
-2. All updates to the forkchoice state resulting from this call **MUST** be made atomically.
+1. All updates to the forkchoice state resulting from this call **MUST** be made atomically.
+
+1. Client software **MUST** return `{status: SUCCESS, payloadId: null}` if `payloadAttributes` is `null`.
 
-3. Client software **MUST** return `{status: SUCCESS, payloadId: null}` if `payloadAttributes` is `null` and the client is not `SYNCING`.
+1. Client software **MUST** return `{status: SYNCING, payloadId: null}` if the payload identified by either the `forkchoiceState.headBlockHash` or the `forkchoiceState.finalizedBlockHash` is unknown. In the event that either the `forkchoiceState.headBlockHash` or the `forkchoiceState.finalizedBlockHash` is unknown, the client software **SHOULD** initiate the sync process.
```

Comment:
Should the **SHOULD** here be a **MUST**?

It's weird to respond with `SYNCING` and not actually kick off some sync process

Reply:
Data supplied by the `forkchoiceUpdated` isn't always enough to initiate the sync process. For instance, there is no state root to pull the state for, so, the **MUST** can't always be satisfied.

Reply:
I see